### PR TITLE
Make sure pycache is full clobbered

### PR DESCRIPTION
**Pull request by jabraham17**
Adds `rm -rf __pycache__` where applicable to make sure we are fully cleaning up generated files when `make clobber` is run.

Tested that `make clobber` deletes the following directories
```
./tools/chpl-language-server/src/__pycache__
./tools/chplcheck/src/__pycache__
./util/test/__pycache__
./util/config/__pycache__
./doc/util/__pycache__
./doc/rst/__pycache__
```

[Reviewed by @DanilaFe]

### CODE REVIEW

File: tools/chpl-language-server/Makefile [hunk 1]
```diff
@@ -44,12 +44,17 @@ test-chpl-language-server: chpl-language-server chpl-language-server-test-venv
...
+clean: clean-link clean-link-shim clean-pycache
 
 cleanall: clean
 
 clobber: clean
 
+clean-pycache:
+	rm -rf $(CHPL_MAKE_HOME)/tools/chpl-language-server/src/__pycache__
+	rm -rf $(CHPL_MAKE_HOME)/tools/chpl-language-server/test/__pycache__
+	rm -rf $(CHPL_MAKE_HOME)/tools/chpl-language-server/test/util/__pycache__
```
Comment (DanilaFe) [inline tools/chpl-language-server/Makefile:56]: could you just use something like `find $(CHPL_MAKE_HOME)/tools/chpl-language-server/ -name "__pycache__" -delete`?

File: tools/chplcheck/Makefile [hunk 2]
```diff
@@ -48,6 +48,9 @@ cleanall: clean
 
 clobber: clean
 
+clean-pycache:
+	rm -rf $(CHPL_MAKE_HOME)/tools/chplcheck/src/__pycache__
```
Comment (DanilaFe) [inline tools/chplcheck/Makefile:52]: Same here.

File: tools/chplcheck/Makefile [hunk 3]
```diff
@@ -49,7 +49,7 @@ cleanall: clean
 clobber: clean
 
 clean-pycache:
-	rm -rf $(CHPL_MAKE_HOME)/tools/chplcheck/src/__pycache__
+		find $(CHPL_MAKE_HOME)/tools/chplcheck -type d -name __pycache__ -exec rm -rf {} +
```
Comment (DanilaFe) [inline tools/chplcheck/Makefile:52]: Seems like an extraneous indent.

**Review by DanilaFe — APPROVED** [review]

File: tools/chplcheck/Makefile (hunk 3 above)
Comment (jabraham17) [reply tools/chplcheck/Makefile]: D'oh

**Comment by bradcray** [comment]
Thanks for picking this up, Jade!

**Pull request merged**